File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/failed_translation/codeforces_637_A.java
import java.util.*;
import java.io.*;

public class codeforces_637_A {
    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(input.readLine());
        String[] s = input.readLine().substring(0, input.readLine().length() - 1).split(" ");
        Map<String, Integer> w = new HashMap<>();
        for (String str : s) {
            w.put(str, w.getOrDefault(str, 0) + 1);
        }
        List<String> reversedS = Arrays.asList(s);
        Collections.reverse(reversedS);
        int c = -1;
        String a = "";
        int maxValue = Collections.max(w.values());
        for (String i : w.keySet()) {
            if (w.get(i) == maxValue) {
                int index = reversedS.indexOf(i);
                if (index > c) {
                    a = i;
                    c = index;
                }
            }
        }
        System.out.println(a);
    }
}

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/java/python/failed_translation/codeforces_637_A.py

import sys
from collections import defaultdict

def main():
    import sys
    input = sys.stdin.read().split()
    n = int(input[0])
    ara = list(map(int, input[1:n+1]))
    map = defaultdict(int)
    for i in range(n):
        v = ara[i]
        map[v] += 1
    max_count = 0
    for count in map.values():
        max_count = max(max_count, count)
    mm = defaultdict(int)
    for i in range(n):
        if mm[ara[i]] == 0:
            mm[ara[i]] += 1
            if mm[ara[i]] == max_count:
                print(ara[i])
                break
        else:
            mm[ara[i]] += 1
            if mm[ara[i]] == max_count:
                print(ara[i])
                break

if __name__ == "__main__":
    main()

